### ISSUE TITLE
Creating RKE cluster with rotateEncryptionKey set to true crashes rancher

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#36333]
**Rancher Server Setup**
- Rancher version: 2.6.3
- Installation option (Docker install/Helm Chart): HA, Hosted Rancher 
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1

**Issue description:**
If you create a custom RKE cluster with rotateEncryptionKey set to true, rancher pods will crash and stay in crash loop back-off

**Business impact:**
This can result in an outage of rancher

**Repro steps:**
 * Create cluster
 * Set a name, click Edit as YAML
 * change rotateEncryptionKey: false to rotateEncryptionKey:true
 * Save
 * Run registration command (worker/etcd/cp) on a fresh VM

**Workaround**
 Is workararound available and implemented? yes
 What is the workaround: edit the clusters.management.cattle.io object and set the rotateEncryptionKey to false.

**Actual behavior:**
rancher crashes, logs:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2a5edfe]goroutine 627123 [running]:
github.com/rancher/rke/cluster.InitClusterObject(0x4cd9e60, 0xc013330c60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00d55eea0, 0x2e, ...)
	/go/pkg/mod/github.com/rancher/rke@v1.3.3/cluster/cluster.go:705 +0xde
github.com/rancher/rke/cmd.RotateEncryptionKey(0x4cd9e60, 0xc013330c60, 0x0, 0xc00c3953a0, 0xc00c3953b0, 0xc017dcc660, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/rancher/rke@v1.3.3/cmd/encryption.go:85 +0x185
github.com/rancher/rke/cmd.ClusterUp(0x4cd9e60, 0xc013330c60, 0xc00c3953a0, 0xc00c3953b0, 0xc017dcc660, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/rancher/rke@v1.3.3/cmd/up.go:107 +0x35b
github.com/rancher/rancher/pkg/kontainer-engine/drivers/rke.clusterUp(0x4cd9e60, 0xc013330c60, 0xc00bbf2578, 0xc00c3953a0, 0xc00c3953b0, 0xc017dcc660, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/rancher/rancher/pkg/kontainer-engine/drivers/rke/rke_driver.go:511 +0x154
github.com/rancher/rancher/pkg/kontainer-engine/drivers/rke.(*Driver).Create(0xc001460f60, 0x4cd9e60, 0xc013330c60, 0xc00ec30bc0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/kontainer-engine/drivers/rke/rke_driver.go:162 +0x37c
github.com/rancher/rancher/pkg/kontainer-engine/types.(*GrpcServer).Create(0xc012378a00, 0x4cd9e60, 0xc013330ba0, 0xc013330bd0, 0xc012378a00, 0xc013330ba0, 0xc018d95ba0)
	/go/src/github.com/rancher/rancher/pkg/kontainer-engine/types/rpc_server.go:42 +0x79
github.com/rancher/rancher/pkg/kontainer-engine/types._Driver_Create_Handler(0x436f440, 0xc012378a00, 0x4cd9e60, 0xc013330ba0, 0xc015014fc0, 0x0, 0x4cd9e60, 0xc013330ba0, 0xc008490000, 0x193e)
	/go/src/github.com/rancher/rancher/pkg/kontainer-engine/types/drivers.pb.go:1437 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00dbeefc0, 0x4d0b8b8, 0xc00d3e3b00, 0xc00d969560, 0xc010a13d40, 0x6ef3b60, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1297 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00dbeefc0, 0x4d0b8b8, 0xc00d3e3b00, 0xc00d969560, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:1626 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00ce1a1a0, 0xc00dbeefc0, 0x4d0b8b8, 0xc00d3e3b00, 0xc00d969560)
	/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:941 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.40.0/server.go:939 +0x1fd
```
**Expected behavior:**

Rancher doesn't crash

 

**Additional notes:**
this might be relevant - https://github.com/rancher/rke/blob/release/v1.3/cmd/up.go#L107

SURE-3986

**Comment by jiaqiluo** [comment]
### Root cause
It is a bug in rancher/rke that RKE will panic if we enable rotate_encryption_key but disable secrets_encryption_config when creating a cluster. The panic happens because RKE tries to use the rkeConfig, which does not exist. The rkeConfig will be made after the rke up command succeeds.

### What was fixed, or what changes have occurred
1. the bug is fixed rancher/rke [GH-2840](https://github.com/rancher/rke/issues/2840)
2. It is not a valid operation to enable the rotateEncryptionKey when the encryption configuration is disabled when creating or updating a cluster. When it happens, rke will either panic(will be fixed) or leave the cluster stuck in updating state and ask users to restore a backup.  That is not a good user experience, so on the Rancher's side, we prevent it from happening during the request validating stage. 

### Areas or cases that should be tested

- create a new RKE cluster, set `rotateEncryptionKey: true` by editing as YAML
- edit an existing RKE  cluster, set `rotateEncryptionKey: true` by editing as YAML

in both cases, the operation is rejected and Rancher UI shows the following message: 

<img width="683" alt="Screen Shot 2022-02-15 at 4 41 42 PM" src="https://user-images.githubusercontent.com/6218999/154170890-fa266f8d-0679-4369-ba43-a49c65b8afd2.png">

- create a new RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML
- edit an existing RKE  cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true`  by editing as YAML

in both cases, the operation should pass
  
### What areas could experience regressions?
generally the areas around the above 4 cases
### Are the repro steps accurate/minimal?
yes

**Comment by timhaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.6-head e4f6f4e
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: RKE Custom cluster

----------

### Testing: ###

Tested this issue with the following:
- Create a new RKE cluster, set `rotateEncryptionKey: true` by editing as YAML
- Edit an existing RKE cluster, set `rotateEncryptionKey: true` by editing as YAML
- Create a new RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML
- Edit an existing RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML

**Result** - **Low Pass**

- Create a new RKE cluster, set `rotateEncryptionKey: true` by editing as YAML - **Pass**
![image.png](https://zube.io/files/rancher/1796e006343489c4eaf6fe79a3becaa2-image.png)
- Edit an existing RKE cluster, set `rotateEncryptionKey: true` by editing as YAML - **Pass**
![image.png](https://zube.io/files/rancher/ac3b46b9de2e0bddc8028f86a6b9bd0f-image.png)
- Create a new RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML **Pass**
- Edit an existing RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML - **Fail**
  - Tested this issue with both custom cluster provisioning and rke node driver provisioning
  - Cluster gets stuck in `Updating` state with the following error
    ![image.png](https://zube.io/files/rancher/533ab7077d257c555c6c4b2b93868f6e-image.png)
  - Provisioning logs all show the following information. The cluster seems to be stuck at `Rotating cluster secrets encryption key`
  - ![image.png](https://zube.io/files/rancher/dd1ef4c5eea0f9b6d87cb4fd3fbac822-image.png)

**Comment by jiaqiluo** [comment]
the fix is available in the latest v2.6-head

**Comment by timhaneunsoo** [comment]
Retesting fixed scenario on v2.6-head 8ece6ae:
- Edit an existing RKE cluster, set `rotateEncryptionKey: true` and `secretsEncryptionConfig.enable: true` by editing as YAML

**Result** - Pass
Cluster goes into `Updating` status with the following message:
![Screen Shot 2022-03-04 at 12.33.06 PM.png](https://zube.io/files/rancher/e6df004cfafe764710d7f894a277aa12-screen-shot-2022-03-04-at-12-33-06-pm.png)

Cluster is successfully updated and returns to `Active` state
![Screen Shot 2022-03-04 at 12.36.30 PM.png](https://zube.io/files/rancher/95b128bc8a7e39219ba13463621a1898-screen-shot-2022-03-04-at-12-36-30-pm.png)
